helm_remote/Tiltfile

```diff
@@ -0,0 +1,47 @@
+# TODO: =====================================
+#   if it ever becomes possible for loaded files to also load their own extensions
+#   this method can be replaced by `load('ext://namespace', 'namespace_create')
+def namespace_create(name):
+    """Returns YAML for a namespace
+    Args:    name: The namespace name. Currently not validated.
+    """
+    k8s_yaml(blob("""apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+""" % name))
+# TODO: end TODO
+#   =====================================
+
+
+def helm_remote(chart, repo_url='', repo_name='', namespace='', values=[], set=[]):
+    if repo_name == '':
+        repo_name = chart
+
+    if namespace == '':
+        namespace = 'default'
+
+    if repo_url != '':
+        local('helm repo add %s %s' % (repo_name, repo_url))  # updates if already added
+
+    if namespace != '':
+        # avoid a namespace not found error
+        namespace_create(namespace)  # do this early so it manages to register before we attempt to install into it
+
+    local('rm -rf helm_deps/%s' % repo_name)
```

Comment:
This requires that `helm_deps` is git ignored right? Do you think we could download these into a temp directory location? or explicitly state these should be ignored. 

This changes the current working directory which feels a little odd.

We currently do 

```
local('if [ ! -d ./tmp/1.40.2/rabbitmq-ha ]; then helm fetch stable/rabbitmq-ha --untar --untardir ./tmp/1.40.2 --version 1.40.2; fi', quiet=QUIET)
rabbitMQYaml = helm('./tmp/1.40.2/rabbitmq-ha',
```
This allows us to cache and version lock the helm charts, but I have never loved that `./tmp` was `.gitignored` but it was part of our setup.

Reply:
That's a good thought about the .gitignore file. It definitely doesn't _require_ that directory to be ignored, but I think it's probably a good idea to create a readme file for this sub-folder which suggests ignoring it.

As far as where the temp dir is located/what it's called. I'm 100% open to suggestions. I could also add an additional function for configuring the temp/cache location? Even so, I'm open to input on what a sane default might be.

Reply:
I agree with it being documented and has sane defaults. 

What about the versioning? Do you think your current method signature would support versioning? (this is something we would love to see and would love to switch to a proper plugin instead of our custom stuff)

Reply:
@drubin as I'm adding the ability to configure the temp/cache location, I'm seeing that your comment was maybe intended to refer to `.tiltignore` rather than `.gitignore`? In that case, yes, you're 100% correct that this dir would need to be ignored in order to prevent infinite recursive triggers to reprocess the tiltfile. So far I haven't been able to find a way around that, yet. Seems like tilt needs a native `unwatch_file()` function for situations like this, or to allow runtime ignoring of dynamic filenames.

Reply:
Added https://github.com/tilt-dev/tilt/issues/3404